base/uk.ac.stfc.isis.ibex.ui.ioccontrol/src/uk/ac/stfc/isis/ibex/ui/ioccontrol/IocPanel.java

```diff
@@ -107,26 +107,30 @@ public void widgetSelected(SelectionEvent e) {
 		collapseButton.addSelectionListener(new SelectionAdapter() {
 			@Override
 			public void widgetSelected(SelectionEvent e) {
-				availableIocsTree.collapseAll();
+				availableIocsTree.getViewer().collapseAll();
 			}
 		});
 		
 		// Add selection tree
-        availableIocsTree = new TreeViewer(this, SWT.FULL_SELECTION);
-        availableIocsTree.setContentProvider(new IOCContentProvider());
-        availableIocsTree.setComparator(new IOCViewerComparator(Comparator.naturalOrder()));
+		Composite treeComposite = new Composite(this, SWT.FILL);
+  		treeComposite.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 1, 1));
+  		treeComposite.setLayout(new GridLayout(1, true));
+
+        availableIocsTree = new FilteredTree(treeComposite, SWT.FULL_SELECTION, new IOCPatternFilter(), true, true);
+        availableIocsTree.getViewer().setContentProvider(new IOCContentProvider());
```

Comment:
Since we end up doing `availableIocsTree.getViewer()` a lot, maybe it's better to just do `final var viewer = availableIocsTree.getViewer()` once just after we create the filtered tree (to reduce duplication)?